unittests/serialization_test_helper.py

```diff
@@ -1,110 +1,27 @@
-                # Recurse into nested models
-                a, c, r = compare_pydantic_models(value1, value2, new_path)
-                added.extend(a)
-                changed.extend(c)
-                removed.extend(r)
-            elif value1 != value2:
-                changed.append((new_path, value1, value2))
-
-    return added, changed, removed
-
-
-def assert_serialization_roundtrip(serializable_object: T) -> T:
-    """
-    Perform a serialization roundtrip and check that the input object is the same after serialization and deserialization.
-
-    :returns the deserialized_object
-    """
-    json_string = serializable_object.model_dump_json()
-    assert json_string is not None
-
-    deserialized_object = serializable_object.model_validate_json(json_data=json_string)
-
-    assert isinstance(deserialized_object, type(serializable_object))
-
-    added, changed, removed = compare_pydantic_models(
-        deserialized_object.segments[0].data_elements[0], serializable_object.segments[0].data_elements[0]
-    )
-
-    assert serializable_object.segments[0].data_elements[0].value_pool is not None
-    assert deserialized_object.segments[0].data_elements[0].value_pool is not None
-
-    serializable_object.segments[0].data_elements[0].value_pool == deserialized_object.segments[0].data_elements[
-        0
-    ].value_pool
-
-    # # Convert Pydantic models to dictionaries
-    # dict1 = serializable_object.model_dump()
-    # dict2 = deserialized_object.model_dump()
-
-    # # Compare the dictionaries
-    # diff = DeepDiff(dict1, dict2, ignore_order=True)
-
-    # # Categorize the differences
-    # added = diff.get("dictionary_item_added", {})
-    # removed = diff.get("dictionary_item_removed", {})
-    # changed = diff.get("values_changed", {})
-
-    # # Output the differences
-    # if added:
-    #     print("Added entries:")
-    #     for key in added:
-    #         print(f"{key}: {added[key]}")
-
-    # if removed:
-    #     print("\nRemoved entries:")
-    #     for key in removed:
-    #         print(f"{key}: {removed[key]}")
-
-    # if changed:
-    #     print("\nChanged entries:")
-    #     for key, value in changed.items():
-    #         print(f"{key}: {value['old_value']} -> {value['new_value']}")
-
-    assert deserialized_object == serializable_object
-    return deserialized_object
+"""
+This module serves a small helper function to test the serialization roundtrip.
+"""
+
+from typing import TypeVar
+
+from pydantic import BaseModel
+
+T = TypeVar("T", bound=BaseModel)
+
+
+def assert_serialization_roundtrip(serializable_object: T) -> T:
+    """
+    Perform a serialization roundtrip and check that the input object is the same after serialization
+    and deserialization.
+
+    :returns the deserialized_object
+    """
+    json_string = serializable_object.model_dump_json()
+    assert json_string is not None
+
+    deserialized_object = serializable_object.model_validate_json(json_data=json_string)
```

Comment:
```suggestion
    deserialized_object = type(serializable_object).model_validate_json(json_data=json_string)
```
Hat sich uns hier ein Fehler eingeschlichen, funktioniert das so?

Reply:
Nevermind, natürlich geht das, hab da iwie nicht richtig nachgedacht ^^